active/0000-prebuild-addons.md

```diff
@@ -0,0 +1,100 @@
+- Start Date: 2017-03-08
+- RFC PR: [#102](https://github.com/ember-cli/rfcs/pull/102)
+
+# Summary
+
+Allow Ember Addons to be "pre-built" such that they produce assets which can be included more efficiently in the build process of a consuming application.
+
+# Motivation
+
+When working on Ember-CLI applications, it is common to include many addons; this is one of the great benefits of the Ember ecosystem. However, in our current system, addons are built with the application they are included in. This means that for each addon you include in your application, you incur a non-trivial impact to your application's build performance.
+
+Since most addon consumers never customize an addon's build or change its source code, it would be beneficial to provide pre-built versions of their assets to improve build performance. Doing so will benefit all applications and allow larger applications, especially those being developed with Engines (which are packaged as addons), to scale to truly ambitious levels.
+
+# Detailed design
+
+## Pre-builds for Addons
+
+A new command will be introduced for use inside of addons only and used as part of the publishing process:
+
+```bash
+ember pre-build
+```
+
+This command will generate assets into a directory named `prebuilt` in the root of the addon. Within `prebuilt` will be a series of directories representing prebuilt assets for different combinations of targets and environments. Since targets can be non-primitive values, they will be hashed for the corresponding directory path.
+
+```bash
+cool-addon/
+  |-prebuilt/
+    |-48e380a165c417253512a904d6b5cf2b-prod/
+    |-48e380a165c417253512a904d6b5cf2b-dev/
+```
+
+Within each of the prebuilt asset directories, there will be directories which contain the results of each `treeFor*` hook in the addon prior to being merged with a consuming application.
+
+```bash
+cool-addon/prebuilt/48e380a165c417253512a904d6b5cf2b-prod/
+  |-addon/
+  |-addon-templates/
+  |-addon-test-support/
+  |-app/
+  |-public/
+  |-styles/
+  |-templates/
```

Comment:
I mean js version of templates generated for `Glimmer1` look something like:
```
Ember.HTMLBars.template((function () {
    return {
      meta: {
        "revision": "Ember@2.8.3",

```
and `Glimmer2`:
```
Ember.HTMLBars.template({ "id": "dxsyziu1", "block": "{...
```
Seems like these 2 formats are not backward compatible?